enhancements/proposed/20210618-tips/tips.md

```diff
@@ -0,0 +1,215 @@
+    minikube dashboard
+    ```
+```
+
+Use `goembed` to embed this file into the minikube binary.
+
+The current `out.Boxed` has a hard-coded style (red). I propose to add another `out.BoxedWithConfig` method to allow
+output with customized style:
+
+```go
+type BoxConfig struct {
+	// box.Config is the config struct of box-cli-maker
+	box.Config
+	// Title is a text that shows as a header of the box
+	Title string
+	// Icon is the optional emoji we want to show as a prefix for the title
+	Icon style.Enum
+}
+
+// BoxedWithConfig writes a templated message in a box with customized style config to stdout
+func BoxedWithConfig(cfg BoxConfig, format string, a ...V) {
+	boxedCommon(String, cfg, format, a...)
+}
+```
+
+Whenever minikube successfully starts, we randomly choose a tip.
+
+Before printing it out, we need to do some regex replacement to strip the markdown syntax
+for better view experience in Terminal:
+
+From this:
+
+``````markdown
+You can specify any Kubernetes version you want. For example:
+
+```
+minikube start --kubernetes-version=v1.19.0
+```
+``````
+
+To this:
+
+```markdown
+You can specify any Kubernetes version you want. For example:
+
+minikube start --kubernetes-version=v1.19.0
+```
+
+Then we can print out the tip:
+
+
+```go
+boxCfg := out.BoxConfig{
+	Config: box.Config{
+		Py: 1,
+		Px: 5,
+		TitlePos: "Top",
+		Type: "Round",
+		Color: tipBoxColor,
+	},
+	Title: tipTitle,
+	Icon: style.Tip,
+}
+
+out.BoxedWithConfig(boxCfg, tips.Tips[chosen] + "\n\n" + tipSuffix)
+```
+
+![Screenshot from 2021-06-18 00-58-02](https://user-images.githubusercontent.com/1311594/122508665-53bd6380-cfd0-11eb-9e99-a6c5935514d5.png)
+
+User can choose to disable this through `minikube config set disable-tips true`
+
+We will have `make generate-docs` generating the docs site based on this YAML file as well.
+
+We can have a `Nice to know` sub-page under `FAQ`?
+
+![Screenshot from 2021-06-18 01-00-30](https://user-images.githubusercontent.com/1311594/122508827-a139d080-cfd0-11eb-98bb-f7c3c1c604c2.png)
+
+
+### About the tip collection
+
+I plan to start with the command lines and cover almost all CLI usages of minikube.
+
+That includes but not limited to:
+- addons
+- cached images
+- command line completion
+- config
+- file copy
+- dashboard
+- delete minikube cluster
+- configure minikube's docker/podman env
+- image build / load / ls / rm
+- ip
+- logging
+- kubectl
+- mount file directory
+- multi-node
+- pause/unpause to save resource
+- multi-profile
+- surface URL to a k8s service
+- ssh into minikube
+- status
+- tunnel to connect to LB
+- update-check to check versions
+- update-context
+
+### Implementation
+
+I plan to open at least 4 PRs:
+
+1. `out.Boxed` with custom style
+2. random `tips` display with ability to disable through config, with an initial set of about 10 tips
+3. `make generate-docs` to sync tips to docs
+4. Add more tips
+
+## Alternatives Considered
+
+1. Is there a more preferred file format to YAML?
```

Comment:
The only other file format we might use is JSON, and while I do like the JSON structure, it has a big flaw which is that new lines must be replaced with \n which can get messy, and make it hard to see how the tip actually looks. I'm pretty happy with YAML for the file format because it makes it very clear how it looks.